google/resource_container_cluster.go

```diff
@@ -785,28 +815,49 @@ func resourceContainerClusterUpdate(d *schema.ResourceData, meta interface{}) er
 
 	lockKey := containerClusterMutexKey(project, zoneName, clusterName)
 
+	var req *container.UpdateClusterRequest
+	var updateDescription string
+	updateF := func() error {
+		var op interface{}
+		location := zoneName
+		switch containerAPIVersion {
+		case v1:
+			op, err = config.clientContainer.Projects.Zones.Clusters.Update(
+				project, zoneName, clusterName, req).Do()
```

Comment:
This change makes me nervous- it feels a bit brittle to me to have this depending on what the value of req is at the time this function gets called. If you'd like to refactor it out, I think wrapping it in another function that takes in req as a parameter and returns this func would be a safer choice.

Reply:
I agree. For that exact reason, I tried going the function route but quickly realised that that number of args to that function became too many and that made me a little uncomfortable. That function would need `config, project, zone, region, isRegionalCluster, request, clusterName, timeout, apiVersion`.
Total of 9 args. But, collapsing zone and region into `location` that number could go down to 8.

Reply:
Ah yeah, I was more just thinking that only `req` would be the parameter since it's the only thing that would change between requests.